Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,233 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <string>
+#include <iostream>
+#include <vector>
+#include "MolSupplier.h"
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+namespace RDKit {
```

Comment:
I think it would make this code easier to use/understand if you add a new namespace under `RDKit`:
```suggestion
namespace RDKit {
namespace GeneralMolSupplier {
```
(you will need to add the `}` to close that namespace at the bottom of the file, of course)